src/controllers/expenses.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable no-shadow */
+'use strict';
+
+const {
+  getAllExpenses,
+  getExpenseById,
+  updateExpenses,
+  removeExpenses,
+  createNewExpense,
+} = require('../services/expenses');
+
+const { getUserById } = require('../services/users');
+
+const addNew = (req, res) => {
+  const expenseData = req.body;
+  const { userId } = expenseData;
+
+  const foundUser = getUserById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExtense = createNewExpense(expenseData);
+
+  res.statusCode = 201;
+  res.send(newExtense);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+
+  const foundExpens = getExpenseById(expenseId);
+
+  if (!foundExpens) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(foundExpens);
+};
+
+const getMany = (req, res) => {
+  const expenses = getAllExpenses();
+  const query = req.query;
+
+  const { userId, category, to, from } = query;
+
+  if (from && to) {
+    const foundExpensesByDate = expenses.filter(
+      (expense) => expense.spentAt > from && expense.spentAt < to
+    );
+
+    res.send(foundExpensesByDate);
+    res.statusCode = 200;
+
+    return;
+  }
+
+  if (category) {
+    const foundExpensesByCategory = expenses.filter(
+      (expense) =>
+        expense.userId === +query.userId
+        && expense.category === query.category
+    );
+
+    res.send(foundExpensesByCategory);
+    res.statusCode = 200;
+
+    return;
+  }
+
+  if (userId) {
+    const foundExpensesByUserId = expenses.filter(
+      (expense) => expense.userId === +userId
+    );
+
+    res.send(foundExpensesByUserId);
+    res.statusCode = 200;
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(expenses);
+};
+
+const remove = (req, res) => {
+  const { expenseId } = req.params;
+
+  const filteredExpenses = getExpenseById(expenseId);
+
+  if (!filteredExpenses) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  removeExpenses(expenseId);
+  res.sendStatus(204);
+};
+
+const update = (req, res) => {
+  const { expenseId } = req.params;
+
+  const foundExpenses = getExpenseById(expenseId);
+
+  if (!foundExpenses) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const { title } = req.body;
+
+  updateExpenses({
+    expenseId,
+    title,
+  });
+
+  res.send(foundExpenses);
```

Comment:
better return updated expense and return it. Now it's an object from an array, but if you will use db, your code won’t work correctly